lib/jxl/decode.cc

```diff
@@ -1073,7 +1079,8 @@ JxlDecoderStatus JxlDecoderProcessInternal(JxlDecoder* dec, const uint8_t* in,
       if (want_preview) {
         if (dec->preview_out_buffer) {
           JxlDecoderStatus status = ConvertImageInternal(
```

Comment:
Are there extra channels in the preview?

Reply:
Yes, the same image metadata applies to it. Outputting those can be added in a later merge request with a function such as `JxlDecoderSetExtraChannelPreviewOutBuffer`